### DISCUSSION TITLE
How to remove last line break, when using Text(text => text.Line())

### DISCUSSION THREAD

**Discussion by joa77** [QuestPDF/QuestPDF#705] [Q&A]
I have an array of strings that I want to render as individual paragraphs.

```csharp
var lines = new string[] { "A", "B", "C" };
page.Text(text =>
{
  foreach(var line in lines)
  {
     text.Line(line);
  }
});
```

Unfortunately, QuestPDF also adds a line break at the end, so that an empty paragraph is created at the end of the text, which takes up unnecessary space.

Is there a way around this?

**Comment by Kazagan** [comment]
I wasn't able to find a way to do it as you have it above.

Alternative way to do it would be using the Column object.

```
var lines = new string[] { "A", "B", "C" };

page.Column(column => 
    {
        column.Item().Text(line);
    }
});
```

The text can still wrap like this though

**Reply by MarcinZiabek** [reply]
As far as I understand, you can use the `text.Span` API to achieve desired result. 

The `text.Line` method concatenates the original text with a new line character.

**Comment by Kazagan** [comment]
Thought of another possibly solution

`Text.Span()` will continue to render on the same line. Could do a for loop
```
for(int i = 0; i < lines.Length - 1; i++)
    text.Line(lines[i]);
    
text.Span(lines[i]);
```

Might need to increment `i` one more time before the `.Span()` responding a bit to early, and mental interpreter is bugging out.

**Reply by MarcinZiabek** [reply]
What about (pseudocode) 😀

```csharp
text.Span(string.Join(Environment.NewLine, lines));
```

or with the `Column` API:
```csharp
.Column(column => 
{
    foreach(var line in lines)
        column.Item().Text(line);
});
```

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
